DEVELOPING_OPAL/validate/src/test/java/org/opalj/fpcf/fixtures/immutability/openworld/assignability/clone_function/SimpleClonePattern.java

```diff
@@ -90,3 +91,16 @@ public ConstructorWithParameter clone(Integer integer){
         return newInstance;
     }
 }
+
+class CloneNonAssignableArrayWithRead {
+
+    @EffectivelyNonAssignableField("field is only assigned once due to the clone function pattern")
+    private boolean[] booleans;
+
+    public CloneNonAssignableArrayWithRead clone(){
+        CloneNonAssignableArrayWithRead newInstance = new CloneNonAssignableArrayWithRead();
+        newInstance.booleans = new boolean[this.booleans.length];
+        boolean dominated = newInstance.booleans[0];
+        return newInstance;
+    }
+}
```

Comment:
Do we want to add another slightly more complex test case, such as the one with the System.arraycopy? To capture cases where values do escape?